src/extension.ts

```diff
@@ -26,26 +28,56 @@ export async function activate(context: vscode.ExtensionContext) {
 	context.subscriptions.push(vscode.commands.registerCommand('azdevops-vscode-simplify.createBranch', (wi: WorkItem) => {
 		wi.createBranch();
 	}));
+	tenantStatusBarItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left);
+	tenantStatusBarItem.command = 'azure-account.selectTenant';
+	context.subscriptions.push(tenantStatusBarItem);
+	context.subscriptions.push(getAzureDevOpsConnection().getAzureAccountExtensionApi().onSessionsChanged(updateTenantStatusBarItem));
+	context.subscriptions.push(getAzureDevOpsConnection().getAzureAccountExtensionApi().onStatusChanged(updateTenantStatusBarItem));
+	context.subscriptions.push(getAzureDevOpsConnection().getAzureAccountExtensionApi().onSubscriptionsChanged(updateTenantStatusBarItem));
+	context.subscriptions.push(getAzureDevOpsConnection().getAzureAccountExtensionApi().onFiltersChanged(updateTenantStatusBarItem));
 }
 
 export function deactivate() { }
 
+async function updateTenantStatusBarItem() {
+	let api = getAzureDevOpsConnection().getAzureAccountExtensionApi();
+	let sessions = api.sessions;
+	const tenants: TenantIdDescription[] = await ((api as any).loginHelper.tenantsTask);
```

Comment:
how can tenants be undefined if it is of type TenantIdDescription[] ? I assume if it can be undefined it should be stated like that, right? Or does a "normal typescript/javascript developer not doing that? :D I learned it by myself as said, so might me that I'm doing it untypically usually
```suggestion
	const tenants: TenantIdDescription[] | undefined = await ((api as any).loginHelper.tenantsTask);
```

Reply:
both good points and I have made the required changes. The tenants variable can indeed not be null (see https://github.com/microsoft/vscode-azure-account/blob/e67ca9bc875b7b1df2d97c00bce932fc7d37ccc5/src/login/AzureAccountLoginHelper.ts#LL57C2-L57C98). And I personally slightly prefer the `!== undefined` check, but we now probably have more checks done in the short way, so let's go with that option